export.py

```diff
@@ -264,6 +273,12 @@ def export_engine(model, im, file, train, half, simplify, workspace=4, verbose=F
         for out in outputs:
             LOGGER.info(f'{prefix}\toutput "{out.name}" with shape {out.shape} and dtype {out.dtype}')
 
+        if dynamic:
+            profile = builder.create_optimization_profile()
+            for inp in inputs:
+                profile.set_shape(inp.name, (1, *im.shape[1:]), (im.shape[0] // 2, *im.shape[1:]), im.shape)
```

Comment:
One small note that I do want to make, though, is that I wasn't sure what to make the optimum input shape into (the 2nd tuple in the parameter list in this line), so I just chose a batch size of half of the max batch size, as there's almost no documentation as to how the optimum batch size should be calculated.

Reply:
Should we set optimum batch size to the --batch-size argument?

Reply:
So then optimum batch size would be the same as the maximum batch size?